AssertionErrors on Alpine Linux

=================================== FAILURES ===================================
________________________________ test_jpg_image ________________________________

    def test_jpg_image():
        file_extension = ".jpg"
        readme_line = test_png_image(file_extension)
>       assert (
            str(readme_line)
            == "![thumbnails-readmeexample](/image_thumbnails/jpg_example_thumb.png)\r\n"
        )
E       AssertionError: assert '![srcexample...humb.png)\r\n' == '![thumbnails...humb.png)\r\n'
E         - ![thumbnails-readmeexample](/image_thumbnails/jpg_example_thumb.png)
E         ?   --------- - ^^^^^
E         + ![srcexample](/image_thumbnails/jpg_example_thumb.png)
E         ?     ^

tests/test_thumbnails_readme.py:106: AssertionError
________________________________ test_svg_image ________________________________

    def test_svg_image():
        file_extension = ".svg"
        readme_line = test_png_image(file_extension)
>       assert (
            str(readme_line)
            == "![thumbnails-readmeexample](/image_thumbnails/svg_example_thumb.png)\r\n"
        )
E       AssertionError: assert '![srcexample...humb.png)\r\n' == '![thumbnails...humb.png)\r\n'
E         - ![thumbnails-readmeexample](/image_thumbnails/svg_example_thumb.png)
E         ?   --------- - ^^^^^
E         + ![srcexample](/image_thumbnails/svg_example_thumb.png)
E         ?     ^

tests/test_thumbnails_readme.py:115: AssertionError
________________________________ test_pdf_image ________________________________

    def test_pdf_image():
        file_extension = ".pdf"
        readme_line = test_png_image(file_extension)
>       assert (
            str(readme_line)
            == "![thumbnails-readmeexample](/image_thumbnails/pdf_example_thumb.png)\r\n"
        )
E       AssertionError: assert '![srcexample...humb.png)\r\n' == '![thumbnails...humb.png)\r\n'
E         - ![thumbnails-readmeexample](/image_thumbnails/pdf_example_thumb.png)
E         ?   --------- - ^^^^^
E         + ![srcexample](/image_thumbnails/pdf_example_thumb.png)
E         ?     ^

tests/test_thumbnails_readme.py:124: AssertionError

--- a/tests/test_thumbnails_readme.py
+++ b/tests/test_thumbnails_readme.py
@@ -3,16 +3,16 @@ import pathlib
 from pathlib import Path
 
 from PIL import Image
-
 from thumbnails_readme import __version__
-from thumbnails_readme.thumbnails_readme import (ImageThumbnail,
-                                                 prepare_readme,
-                                                 prepare_thumbnails_folder)
+from thumbnails_readme.thumbnails_readme import (
+    ImageThumbnail,
+    prepare_readme,
+    prepare_thumbnails_folder,
+)
 
 # Workaround for Windows cairosvg import error
 # can use GIMP / Inkscape instead
 # os.environ["path"] += r";C:\Program Files\UniConvertor-2.0rc5\dlls"
-
 
 TESTFILE_CONTENTS_BEFORE = """# thumbnails-readme --- Create thumbnails\n\n---"""
 
@@ -56,7 +56,6 @@ def test_readme_creation():
 
     assert os.path.exists("./___README.md") == 1
 
-
 def test_readme_preparation():
     prepare_readme("./___README.md")
     with open("./___README.md", "r") as readme:
@@ -77,8 +76,8 @@ def create_jpg_image():
 
 
 # Test default png, works with different extensions: jpg, svg, pdf
-def test_png_image(file_extension=".png"):
-    file = "example" + file_extension
+def test_png_image():
+    file = "example.png"
     image = ImageThumbnail(
         Path(path),
         os.path.splitext(file)[0],
@@ -97,34 +96,105 @@ def test_png_image(file_extension=".png"):
         + "_thumb.png"
         + ")\r\n"
     )
-    return readme_line
+    assert (
+        str(readme_line)
+        == "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + "example](/image_thumbnails/png_example_thumb.png)\r\n"
+    )
 
 
 def test_jpg_image():
-    file_extension = ".jpg"
-    readme_line = test_png_image(file_extension)
+    file = "example.jpg"
+    image = ImageThumbnail(
+        Path(path),
+        os.path.splitext(file)[0],
+        os.path.splitext(file)[1],
+        MAX_SIZE,
+        pdf_quality,
+    )
+    readme_line = (
+        "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + image.file_name
+        + "](/image_thumbnails/"
+        + image.file_extension.split(".")[-1].lower()
+        + "_"
+        + image.file_name
+        + "_thumb.png"
+        + ")\r\n"
+    )
     assert (
         str(readme_line)
-        == "![thumbnails-readmeexample](/image_thumbnails/jpg_example_thumb.png)\r\n"
+        == "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + "example](/image_thumbnails/jpg_example_thumb.png)\r\n"
     )
 
 
 def test_svg_image():
-    file_extension = ".svg"
-    readme_line = test_png_image(file_extension)
+    file = "example.svg"
+    image = ImageThumbnail(
+        Path(path),
+        os.path.splitext(file)[0],
+        os.path.splitext(file)[1],
+        MAX_SIZE,
+        pdf_quality,
+    )
+    readme_line = (
+        "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + image.file_name
+        + "](/image_thumbnails/"
+        + image.file_extension.split(".")[-1].lower()
+        + "_"
+        + image.file_name
+        + "_thumb.png"
+        + ")\r\n"
+    )
     assert (
         str(readme_line)
-        == "![thumbnails-readmeexample](/image_thumbnails/svg_example_thumb.png)\r\n"
+        == "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + "example](/image_thumbnails/svg_example_thumb.png)\r\n"
     )
 
 
 def test_pdf_image():
-    file_extension = ".pdf"
-    readme_line = test_png_image(file_extension)
+    file = "example.pdf"
+    image = ImageThumbnail(
+        Path(path),
+        os.path.splitext(file)[0],
+        os.path.splitext(file)[1],
+        MAX_SIZE,
+        pdf_quality,
+    )
+    readme_line = (
+        "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + image.file_name
+        + "](/image_thumbnails/"
+        + image.file_extension.split(".")[-1].lower()
+        + "_"
+        + image.file_name
+        + "_thumb.png"
+        + ")\r\n"
+    )
     assert (
         str(readme_line)
-        == "![thumbnails-readmeexample](/image_thumbnails/pdf_example_thumb.png)\r\n"
+        == "!["
+        + pathlib.PurePath(image.path_to_file).parent.name
+        + "example](/image_thumbnails/pdf_example_thumb.png)\r\n"
     )
+
+
+def cleanup():
+    os.remove("./examplefile.png")
+    assert os.path.exists("./examplefile.png") == 0
+    os.remove("./examplefile.jpg")
+    assert os.path.exists("./examplefile.jpg") == 0
+    os.remove("./___README.md")
+    assert os.path.exists("./___README.md") == 0
 
 
 # Due to windows path issues, the tests are not run on windows

--- a/thumbnails_readme/thumbnails_readme.py
+++ b/thumbnails_readme/thumbnails_readme.py
@@ -7,15 +7,12 @@ from pdf2image import convert_from_path
 from PIL import Image
 import cairosvg
 
-
 class ImageThumbnail:
     """
     Class for the image
     """
 
-    def __init__(
-        self, path_to_file, file_name, file_extension, max_size, pdf_quality
-    ):
+    def __init__(self, path_to_file, file_name, file_extension, max_size, pdf_quality):
         """
         Default constructor
         """
@@ -45,25 +42,31 @@ class ImageThumbnail:
         if os.name == "nt":
             # Windows - poppler path required
             images = convert_from_path(
-                self.path_to_file, self.pdf_quality, poppler_path=poppler_path, size=self.max_size[0]
+                self.path_to_file,
+                self.pdf_quality,
+                poppler_path=poppler_path,
+                size=self.max_size[0],
             )
         else:
             # Linux - sudo apt-get install poppler-utils
-            images = convert_from_path(self.path_to_file, self.pdf_quality, size=self.max_size[0])
+            images = convert_from_path(
+                self.path_to_file, self.pdf_quality, size=self.max_size[0]
+            )
         for image in images:
             image.save(
-                f"{path_to_thumbnails_folder}"
-                f"/pdf_{self.file_name}_thumb.png"
+                f"{path_to_thumbnails_folder}" f"/pdf_{self.file_name}_thumb.png"
             )
+
     def create_svg_thumbnail(self, path_to_thumbnails_folder):
         """
         Create thumbnail for the SVG
         """
-        cairosvg.svg2png(url=str(self.path_to_file), output_width=self.max_size[0], output_height=self.max_size[1],
-                         write_to=
-            f"{path_to_thumbnails_folder}"
-            f"/svg_{self.file_name}_thumb.png"
-         )
+        cairosvg.svg2png(
+            url=str(self.path_to_file),
+            output_width=self.max_size[0],
+            output_height=self.max_size[1],
+            write_to=f"{path_to_thumbnails_folder}" f"/svg_{self.file_name}_thumb.png",
+        )
 
     def write_to_readme(self, readme):
         """
@@ -90,13 +93,9 @@ def prepare_readme(path_to_readme):
         split_filename_extension[0] + "_temp" + split_filename_extension[1]
     )
 
-    with open(path_to_readme) as readme, open(
-        path_to_readme_temp, "w"
-    ) as readme_two:
+    with open(path_to_readme) as readme, open(path_to_readme_temp, "w") as readme_two:
         for line in readme:
-            if not any(
-                remove_line in line for remove_line in lines_for_removal
-            ):
+            if not any(remove_line in line for remove_line in lines_for_removal):
                 readme_two.write(line)
         readme.close()
         readme_two.close()
@@ -148,9 +147,7 @@ def crawl(
                 if image.file_extension.lower() in [".jpg", ".jpeg", ".png", ".gif"]:
                     image.create_raster_thumbnail(path_to_thumbnails_folder)
                 elif image.file_extension.lower() in [".pdf"]:
-                    image.create_pdf_thumbnail(
-                        path_to_thumbnails_folder, poppler_path
-                    )
+                    image.create_pdf_thumbnail(path_to_thumbnails_folder, poppler_path)
                 elif image.file_extension.lower() in [".svg"]:
                     image.create_svg_thumbnail(path_to_thumbnails_folder)
                 if image.file_extension.lower() in [

